src/player_activity.h

```diff
@@ -29,10 +31,169 @@ class player;
 class translation;
 class activity_ptr;
 
+struct simple_task {
+    // Name of the target that's being processed
+    const std::string target_name;
+    /* Total number of moves required for this target/task to complete */
+    const int moves_total = 0;
+    /* The number of moves remaining for this target/task to complete */
+    int moves_left = 0;
+
+    simple_task( const std::string &name_, int moves_total_ )
+        : target_name( name_ ), moves_total( moves_total_ ), moves_left( moves_total_ ) {
+    }
+    simple_task( const std::string &name_, int moves_total_, int moves_left_ )
+        : target_name( name_ ), moves_total( moves_total_ ), moves_left( moves_left_ ) {
+    }
+
+    inline const bool complete() const {
+        return moves_left <= 0;
+    }
+
+    //Json stuff
+
+    void serialize( JsonOut &json ) const;
+
+    simple_task() = default;
```

Comment:
```suggestion
    inline const bool complete() const {
        return moves_left <= 0;
    }

    //Json stuff

    void serialize( JsonOut &json ) const;
```
these constructors doesn't seem to be necessary

Reply:
wdym, those are called multiple times in a code, second one is only for deserialize, but yeah, deafult() is not necessary

Reply:
unless you need to run validation on struct construction, we'd probably don't need these:

> moves_total and target_name are constants, so u can't modify those outside constructor

it's possible using [aggregate initialization](ithttps://en.cppreference.com/w/cpp/language/aggregate_initialization), check [godbolt](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIMwAcpK4AMngMmAByPgBGmMQgAMyJpAAOqAqETgwe3r7%2BQemZjgJhEdEscQnJtpj2JQxCBEzEBLk%2BfoG19dlNLQRlUbHxSSkKza3t%2BV3j/YMVVaMAlLaoXsTI7BzmieHI3lgA1CaJbuPE4cAn2CYaAII7eweYx6dO55is17cP9%2BdeDkOmRYqXoAH1mgoANbHADsVnuhyRhzQDHGQII6BAIHOl0OE2AmAIYOYbBOCLuyJRAnR4QIhxYqAAbpgFBDUM1aK8ACKHDTkn5UukM5mssH0Kj0k68/mJCmC5HhWjhF6o9ExVCeakg%2BgETAQJbUtFS%2BEKqlI4hE9YMEUstkSqWnaV8gWI5EmWHcn5mw5UC6uDFYkAZAiWr5mABsh1QqXiTCIxBObiTEHGQZDYZY5ijGVIRvRwNBmAhTGh2fxpahho9FPNyIyr2Tp2OZjMhfBkJhNYrxEJxNJ7GOAFZk63zGZG0mK9CAHQEokk1iYH11yfN8cmEfjvOMu3szkgFsTpNTzsz3dioicld1k/r1s70X2zCSw/jtduadQ89P8Uvgg3uad6ftu2pFnqb6th%2BhwQGeaA6kS%2BrVokABiR6hl4y5QZBZhUGISjjshTYgVBHpemOcqAYcloENa0YKK6lLup63osZR9w/EwXhEAyTDhAahwALTXIcwo1j6ar0lxPGdlwPJAngCElmWppunWc4tH2i5sPJ46yduVE/nuV5iPJXAaBopCGRez6SmZlk%2BmRjFUpJhzSagX7HokvLtsWnbiWp5oab2C4DrpradmYBmBU57G/ExSJptiaDcdBsnQUlICuLQznIplKWOsRnkZZi2LZYxPo0XRsryix9wcCstCcEOvB%2BBwWikKgnCjlYlhAmsGwvDsPCkAQmgNSsUKjDOEZmBoXAAJxDhoEZDgtsIBGY%2BicJIrXjZ1nC8AoICWWN7UNaQcCwEg8GpHQ8TkJQt33QkwBSFtNC0HqxDHRAMT7TE4QtAAnpwI2A8wxDAwA8jE2iYA4YO8PBbCCNDDC0KD52kFgMReMAbhiLQx3cLwWAsIYwDiNj%2BCWg4eB2vtmCqAj3FbB1dJ1PtyoxMQIMeFg%2B2hopSOkCyxAako3KYBTRjKkY40rFQBjAAoABqeCYAA7tDsZtSN/CCCIYjsFIMiCIoKjqNjuhcPolMoNY1j6HgMTHZAKwxg0JOCYJabSqYljWHNvCisQFxYO7BrdAjDQuAw7ieB0eihCqCwjHbRRZAIUx%2BJnGTZww8zDAkdt2LHvQTG0Sf5GXdQVwIfStMXlQZ7YVe53oszN2nJcSCsCgDZs/fbRwLWkG1HVdRwhyqAEEaCRGkiHMAyDIIcUgzhOEC4IQJAtokXBLLwZ1aEsk3TRokhL7CXARhZEYBAEkiwqPu2kCwEgWRP%2B3T0dJ2jUVpdGAiBgzcVSNxR6EBnr0GIJEJcnA54LyXivNeG9JBb14JgfACY8BYjtobYQohxBm0IZbNQ%2B1bakC1nzVISNGrNT2tjae0NwGpVQFQWe89F7L1XuvTe28PAghegfI%2BJ9FYX2SDOZIMjZFyLfkwqeh1bAANPhNUeZhFGh2UWo8%2BYt4iZGcJIIAA%3D%3D%3D)

Reply:
Said it in kinda wrong way, those two are implicitly called at emplace methods of progress_counter, I'd point that by code block, but I don't know how